opm/grid/common/WellConnections.cpp

```diff
@@ -82,13 +82,23 @@ WellConnections::WellConnections(const std::vector<OpmWellType>& wells,
                                  const std::unordered_map<std::string, std::set<int>>& possibleFutureConnections,
                                  const Dune::CpGrid& cpGrid)
 {
-    const auto& cpgdim = cpGrid.logicalCartesianSize();
+    // - If the grid with dimensions {NX, NY, NZ} has been globally refined 'count' times,
+    //   then cpgrid.logicalCartesianSize() returns {2^count * NX, 2^count * NY, 2^count * NZ}.
+    //   By default, global refinement subdivides each cell into 2 in each direction (x, y, and z).
+    //
+    // - If the grid has undergone local (not global) refinement, cpgrid.logicalCartesianSize()
+    //   returns the logical Cartesian size of the original (level-zero) grid.
+    const auto& cpgdim = cpGrid.maxLevel()? cpGrid.currentData().front()->logicalCartesianSize() : cpGrid.logicalCartesianSize();
+
     // create compressed lookup from cartesian.
     std::vector<int> cartesian_to_compressed(cpgdim[0]*cpgdim[1]*cpgdim[2], -1);
 
-    for( int i=0; i < cpGrid.numCells(); ++i )
+    // If the grid has LGRs, use globalCell from level zero grid.
+    const auto& globalCell = cpGrid.maxLevel()? cpGrid.currentData().front()->globalCell() : cpGrid.globalCell();
```

Comment:
Same here: we seem to do the same in both branches. If there is only one level  `cpGrid.globalCell()` and `cpGrid.currentData().front()->globalCell()` yield the same.

Reply:
Thanks, both comments have been addressed.